### PR TITLE
move chat attachment download path logic to Go

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/actions/platform-specific/index.native.tsx [hunk 1]
```diff
@@ -646,6 +646,14 @@ export const watchPositionForMap = async (errFn: () => void): Promise<number> =>
   return watchID
 }
 
+const configureFileAttachmentDownloadForAndroid = () =>
+  RPCChatTypes.localConfigureFileAttachmentDownloadLocalRpcPromise({
+    // Android's cache dir is (when I tried) [app]/cache but Go side uses
+    // [app]/.cache by default, which can't be used for sharing to other apps.
+    cacheDirOverride: RNFetchBlob.fs.dirs.CacheDir,
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.native.tsx]: @MarcoPolo this ok

Comment (songgao) [reply shared/actions/platform-specific/index.native.tsx]: Some more context: this is actually the same path that we've been using for Android download. I just plumbed this into Go side. Fs uses the same cache dir.

File: go/chat/attachments/downloader.go [hunk 2]
```diff
@@ -15,54 +14,42 @@ import (
...
-	} else {
-		if err := libkb.MakeParentDirs(g.Log, filename); err != nil {
-			return "", nil, err
-		}
+	if err := os.MkdirAll(parentDir, libkb.PermDir); err != nil {
+		return "", nil, err
+	}
+
+	reason := chat1.GetThreadReason_GENERAL
+	unboxed, err := g.ChatHelper.GetMessages(ctx, uid, convID,
```
Comment (joshblum) [inline go/chat/attachments/downloader.go]: Preexisting but you switch this to use `ChatHelper.GetMessage` instead? Slightly cleaner and does a bounds check on the array before accessing.

**Comment by songgao** [comment]
CI error seems irrelevant; merging!

**Pull request merged**